src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -91,32 +90,43 @@ def __init__(
         self._step_details = step_details
         self._logger = LoggerAdapter(OPENJD_LOG, extra={"session_id": session_id})
 
-    def set_step_script(self, manifests, path_mapping, s3_settings) -> None:
-        # TODO - update to run python as embedded file
-        profile = os.environ.get("AWS_PROFILE")
-        deadline_path = os.path.join(Path(sysconfig.get_path("scripts")), "deadline")
+    def set_step_script(self, manifests, s3_settings) -> None:
+        """Sets the step script for the action
 
-        script = "#!/usr/bin/env bash\n\n{} attachment download -m {} --path-mapping-rules {} --s3-root-uri {}  --profile {}".format(
-            deadline_path,
-            " -m ".join(quote(v) for v in manifests),
-            quote(path_mapping),
+        Parameters
+        ----------
+        manifests : list[BaseAssetManifest]
+            The job attachment manifests
+        s3_settings : JobAttachmentS3Settings
+            The job attachment S3 settings
+        """
+        args = [
+            "{{ Task.File.AttachmentDownload }}",
+            "-pm",
+            "{{ Session.PathMappingRulesFile }}",
+            "-s3",
             s3_settings.to_s3_root_uri(),
-            profile,
-        )
-
-        self._step_script = StepScript_2023_09(
-            actions=StepActions_2023_09(
-                onRun=Action_2023_09(command="{{ Task.File.AttachmentDownload }}")
-            ),
-            embeddedFiles=[
-                EmbeddedFileText_2023_09(
-                    name="AttachmentDownload",
-                    type=EmbeddedFileTypes_2023_09.TEXT,
-                    runnable=True,
-                    data=script,
-                )
-            ],
-        )
+            "-m",
+        ]
+        args.extend([quote(p) for p in manifests])
```

Comment:
We shouldn't need to use `quote(p)` here since OpenJD step script commands and arguments are not treated as shell commands.

```suggestion
        args.extend(manifests)
```

Reply:
Good catch, updated.